web_src/js/features/comp/ReactionSelector.js

```diff
@@ -1,38 +1,42 @@
 import $ from 'jquery';
 import {POST} from '../../modules/fetch.js';
 
-export function initCompReactionSelector($parent) {
-  $parent.find(`.select-reaction .item.reaction, .comment-reaction-button`).on('click', async function (e) {
-    e.preventDefault();
+export function initCompReactionSelector() {
+  for (const container of document.querySelectorAll('.comment-list, .diff-file-body')) {
+    container.addEventListener('click', async (e) => {
+      const item = e.target.closest('.item.reaction');
+      const button = e.target.closest('.comment-reaction-button');
+      if (!item && !button) return;
+      e.preventDefault();
 
-    if (this.classList.contains('disabled')) return;
+      const target = item || button;
+      if (target.classList.contains('disabled')) return;
 
-    const actionUrl = this.closest('[data-action-url]')?.getAttribute('data-action-url');
-    const reactionContent = this.getAttribute('data-reaction-content');
-    const hasReacted = this.closest('.ui.segment.reactions')?.querySelector(`a[data-reaction-content="${reactionContent}"]`)?.getAttribute('data-has-reacted') === 'true';
+      const actionUrl = target.closest('[data-action-url]').getAttribute('data-action-url');
+      const reactionContent = target.getAttribute('data-reaction-content');
+      const hasReacted = target.closest('.segment.reactions').querySelector(`a[data-reaction-content="${CSS.escape(reactionContent)}"]`).getAttribute('data-has-reacted') === 'true';
```

Comment:
Here seems not right (maybe a bug introduced by previous PRs)

It should `closet('.content').find('.segment.reactions')`

Reply:
I think it could be improved by using "comment-container".

If you don't mind, I would try to rewrite some related code and add some comments.

Reply:
I already fixed this by using `target.closest('.comment').querySelector('.segment.reactions')` which is a common parent present on all three types of comments.

Reply:
Feel free to do more refactors now.

Reply:
BTW there is one unknown behaviour left: If we move `const content = target.closest('.content');` right after `if (data`, it will error sometimes. I don't know why that is and it works when having it before the POST.

Reply:
I have hugely simplified the code